test/integration/qrCodeAuth.test.js

```diff
@@ -48,21 +51,39 @@ describe("QrCodeAuth", function () {
         });
     });
 
-    it("Should return a 500 status code and the correct error message when an error occurs while storing user device info", function (done) {
+    it("should fail with 404, when the user is not found", function (done) {
       chai
         .request(app)
         .post("/auth/qr-code-auth")
         .send(wrongUserDeviceInfoData)
         .end((err, res) => {
           if (err) {
-            return done();
+            return done(err);
+          }
+
+          expect(res).to.have.status(404);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Document not found!");
+          expect(res.body.error).to.equal("Not Found");
```

Comment:
If the failure is due to the user not being found, could we provide a more precise response? Using "Document not found" might imply that the QR code document is missing, whereas the actual issue is the user doesn't exist.

Reply:
addressed this